### DISCUSSION TITLE
Termination event Experiment

### DISCUSSION THREAD

**Discussion by sklein-ise** [pybamm-team/PyBaMM#1796] [Q&A]
Hello again,

Would it be possible to add more termination options to the experiment class? I wanted to simulate a GITT experiment for the half cell model and terminate the experiment at a certain voltage limit (0.01). But right now "Only capacity can be provided as a termination reason". 

```
experiment = pybamm.Experiment(
    [(
        "Discharge at C/20 for 1 second or until 0.01 V (0.001 seconds period)", 
        "Discharge at C/20 for 20 seconds or until 0.01 V (0.01 seconds period)", 
        "Discharge at C/20 for 1419 seconds or until 0.01 V (1 seconds period)", 
        "Rest for 1 seconds (0.01 seconds period)",
        "Rest for 10 seconds (0.1 seconds period)",
        "Rest for 10789 seconds (1 minute period)",
    
    )] * 60
)
```

I was trying to find a workaround by adding a condition for the Rest period but it didn't work. Is there another way?

Thanks in advance!

**Comment by brosaplanella** [comment]
We can certainly add more termination events (feel free to open an issue with some ideas on other termination options to include). Though you can probably pull it from the existing functionality. Why do you want to add the condition for the rest period? If you do GITT discharging you will hit the critical voltage during discharge rather than rest.

**Reply by sklein-ise** [reply]
Right now, when the experiment hits the critical voltage at the discharge step, it doesn't terminate but jumps to the next Rest step. After relaxing it will do discharge steps again. This keeps going until all cycles are done or the simulation fails. So I have to know the exact number of cycles beforehand. But I wanted to fit parameters to a GITT experiment and the number of cycles are varying.

**Comment by brosaplanella** [comment]
I see! Yes, that would be a nice thing. Could you please open an issue about that? It would be nice to think about other possible termination criteria, such as `SoC`.

**Reply by Brushnie** [reply]
I’m wondering if this has been implemented or not to set the termination to be SOC. And is there a document where I can find for all the available termination criteria?

**Reply by Brushnie** [reply]
@brosaplanella hello, is there an issue opened for this? Is it possible to simulate cycling between fixed SOC windows?

**Reply by brosaplanella** [reply]
There is this issue open: https://github.com/pybamm-team/PyBaMM/issues/2553

But that's work in progress currently.
